yaptide/celery/utils/manage_tasks.py

```diff
@@ -87,3 +92,58 @@ def translate_celery_state_naming(job_state: str) -> str:
         return EntityState.COMPLETED.value
     # Others are the same
     return job_state
+
+
+def handle_shieldhit_cancellation(tasks: list[CeleryTaskModel]):
+    """Function triggering shieldhit processes to terminate and dump results"""
+    command_as_list = ['kill', '--signal', 'SIGINT']
+    celery_ids = []
+    for task in tasks:
+        if task.task_state == EntityState.RUNNING.value:
+            logging.warning(task.sim_pid)
+            command_as_list.append(str(task.sim_pid))
+        elif task.task_state in (EntityState.PENDING.value, EntityState.UNKNOWN.value) or task.sim_pid is None:
+            celery_ids.append(task.celery_id)
+
+    # Send SIGINT to shieldhit processes, that will immediately stop the simulation and trigger output files dump
+    subprocess.run(command_as_list, check=True)
```

Comment:
it seems better option is to use `os.kill(pid, signal.SIGTERM)` as `kill` may not be available on docker python image

Reply:
There is another thing - what if we deploy yaptide in such way that there are couple of celery workers, running on different VMs and communicating by network via redis. 
Then some of the parallel tasks would run on machine A and some on machine B.
Flask itself would run on machine C.

There is no way to kill a process on machine B, being on machine C.